### PR TITLE
Removed unused udeps, workflow to find unused dependencies

### LINKED ISSUE #1213: Run cargo udeps on CI
Tool: https://github.com/est31/cargo-udeps

I think we can run this once a week to alert us on unused dependencies. It might also be nice to run it on PRs so we get notified in cases we no longer need a dependency.

There's a github action for it: https://github.com/aig787/cargo-udeps-action

### PR DESCRIPTION
**Pull request by nomaxg**
closes #1213

### CODE REVIEW

File: .github/workflows/contracts.yml [hunk 1]
```diff
@@ -21,6 +21,7 @@ env:
 
 jobs:
   contracts:
+    timeout-minutes: 60
```
Comment (nomaxg) [inline .github/workflows/contracts.yml:24]: @mrain @alxiong is this a reasonable timeout for this job?

Comment (mrain) [reply .github/workflows/contracts.yml:24]: cc @alysiahuggins @philippecamacho do you have more inputs?

Comment (nomaxg) [reply .github/workflows/contracts.yml:24]: Actually I think this is more than reasonable, anything longer than an hour shouldn't block CI (an hour is arguably pretty generous already).

Comment (mrain) [reply .github/workflows/contracts.yml:24]: Right, I think it's more than enough

Comment (tbro) [reply .github/workflows/contracts.yml:24]: Looking at healthy tests I see they pass in ~11min, can we reduce this to ~15 or ~20?

Comment (tbro) [reply .github/workflows/contracts.yml:24]: Here you can see they consistently succeed in <= 12 min. https://github.com/EspressoSystems/espresso-sequencer/actions/workflows/contracts.yml

Comment (nomaxg) [reply .github/workflows/contracts.yml]: done https://github.com/EspressoSystems/espresso-sequencer/pull/1216/commits/5ea2b330911d7949b83d7d12bd2f50c9863e1209

**Review by tbro — APPROVED** [review]
:+1:

**Pull request merged**